lib/srv/desktop/windows_server.go

```diff
@@ -591,6 +594,7 @@ func (s *WindowsService) initializeLDAP() error {
 	}
 
 	conn.SetTimeout(ldapRequestTimeout)
+	s.lc.SetConnectionCreator(createLDAPConnection)
```

Comment:
Why do we need this connection creator thing? (Instead of just using ldap.Dial directly)

Reply:
We set up TLS here but we create new connection inside `ldap.ReadWithFilter` (when we get new referral to check in the response). This way we have factory for these connection with certificates and so on

Reply:
AFAICT it's called only in `WindowsService.initializeLDAP`, so is `SetClient`. Feels like `s.lc` could be created here instead and then we could avoid those setters and mutex. `LDAPConfig` is available in `s.cfg.LDAPConfig` anyway. Chances are I'm missing something here.

Reply:
We pass `s.lc` to in `NewWindowsService` (`lib/srv/desktop/windows_server.go:387`) before `initializeLDAP`. We could probably rework that but it'd be awkward. I'd rather leave it as is for now.